crates/committed/src/checks.rs

```diff
@@ -326,6 +327,14 @@ pub(crate) fn check_fixup(
     }
 }
 
+pub(crate) fn strip_fixup(message: &str) -> &str {
+    if let Some(message) = message.strip_prefix("fixup! ") {
```

Comment:
This has a semantic conflict with #395

Reply:
I'm sorry, I was targeting the main branch when creating this, and therefore ensured that semantics are consistent with the target. It would be non-atomic to have the work on this PR rely on another as-yet unmerged PR.

Reply:
Posting multiple PRs that will conflict when merged in non-obvious ways is not helpful and is depending on the maintainer to catch something that was known.

In a case like this, you can always wait.

Reply:
While I understand that this PR has a semantic conflict with #395, I have had no indication of whether #395 will be merged. Therefore, it seemed best to ensure this PR aligns with the current state of the `master` branch. 

In my experience, it's not uncommon to have PRs that may need adjustments as other related PRs are merged. This ensures that each change is reviewed and integrated based on its own merits and the current state of the target branch. Furthermore, having a PR be created for visibility is, in my experience, better than relying on contributors waiting before adding further improvements.

I also split up my PRs based on your earlier feedback. The boundary between "atomic" commits and PRs can sometimes be nuanced. The functionality to strip the prefix is distinct from the changes introduced in #395, which is why they were separated.

If or when it comes to merging this PR in due time, the necessary adjustments can be made based on the state of the `master` branch at that time.

Reply:
> In my experience, it's not uncommon to have PRs that may need adjustments as other related PRs are merged. 

I understand that and don't have a problem with it.  What I am concerned about is the lack of communication on that front.  The relationship of the PRs needs to be made clear.

> Furthermore, having a PR be created for visibility is, in my experience, better than relying on contributors waiting before adding further improvements.

There are multiple ways of handling that.  One is splitting where you can, like you did here.  The other is communicating it.  You did that by talking about this in the issue and you can foreshadow that in a PR description.